### ISSUE TITLE
[Feature request] Support more  partiticaption status like needs-action

### ISSUE THREAD

**Issue by Thaodan** [nextcloud/tasks#2135]
### Expected behaviour

Nextcloud tasks should support more participation statuses such as needs action or delegated (#72).
The UI should show something  like `Status: <Status>`

The usage of such an extended status view would be to be able to visualize the status of a task better.
This includes things such as:  blocked by another task, waiting and so on.

### Actual behaviour

The only supported state right now is the done or undone state portrait by the checkbox next to the taskname.

### Server configuration

**Nextcloud version:** 24.0.4

**Tasks version:**  0.14.4-2

**List of activated apps:**

```
Not relevant
```

**Nextcloud configuration:**
Not relevant

## See also:
- [RFC2445 4.2.12 Participation Status](https://www.ietf.org/rfc/rfc2445.txt)

**Comment by raimund-schluessler** [comment]
The app already has the possibility to set the status of a task in the right sidebar:
![grafik](https://user-images.githubusercontent.com/2496460/233423168-462a758a-4aa3-42b2-9e7f-b387ba68c3d0.png)

We offer the following choices for the status:
https://github.com/nextcloud/tasks/blob/7536446b036bc1f479b9d0010f95b4238639183e/src/views/AppSidebar.vue#L423-L448 according to RFC 5545, section 3.8.1.11 https://datatracker.ietf.org/doc/html/rfc5545#section-3.8.1.11

In order to support the Participation status according to https://datatracker.ietf.org/doc/html/rfc5545#section-3.2.12 as you propose, we would first need to implement attendees. Without the option to set or invite attendees, it does not make sense to implement a participation status.

**Comment by PeonyThePeon** [comment]
"Needs action" should probably be automatic, and should be equivalent to the [RELTYPE=DEPENDS-ON parameter from RFC 9253](https://www.rfc-editor.org/info/rfc9253/#name-additional-new-reltype-para) not being populated.

In other words, if a task is not currently blocked, then by definition it needs action.

See https://github.com/nextcloud/tasks/issues/3188
